alpine/Dockerfile

```diff
@@ -6,33 +6,87 @@ ARG TOOLBOX_EXCLUDED_TAGS="exclude-php:${PHP_VERSION}"
 ARG TOOLBOX_VERSION="1.62.0"
 
 FROM ${BASE_IMAGE} as phpbase
-
 LABEL maintainer="Jakub Zalas <jakub@zalas.pl>"
-ARG BUILD_DEPS="autoconf file g++ gcc libc-dev pkgconf re2c"
 ARG LIB_DEPS="zlib-dev libzip-dev bzip2-dev icu-dev"
 ARG TOOL_DEPS="git graphviz ttf-freefont make unzip gpgme gnupg-dirmngr openssh-client"
+RUN --mount=type=cache,target=/var/cache/apk,sharing=locked,id=apk \
+  apk add --no-cache $TOOL_DEPS $LIB_DEPS
 
-RUN apk add --no-cache --virtual .tool-deps $TOOL_DEPS $LIB_DEPS \
- && apk add --no-cache --virtual .build-deps $BUILD_DEPS \
- && git clone https://github.com/nikic/php-ast.git && cd php-ast && phpize && ./configure && make && make install && cd .. && rm -rf php-ast && docker-php-ext-enable ast \
- && pecl install pcov && docker-php-ext-enable pcov \
- && docker-php-ext-configure intl \
- && docker-php-ext-install zip pcntl bz2 intl bcmath \
- && echo "date.timezone=Europe/London" >> $PHP_INI_DIR/php.ini \
- && echo "memory_limit=-1" >> $PHP_INI_DIR/php.ini \
- && echo "phar.readonly=0" >> $PHP_INI_DIR/php.ini \
- && echo "pcov.enabled=0" >> $PHP_INI_DIR/php.ini \
- && apk del .build-deps
+FROM phpbase AS builder-base
+ARG BUILD_DEPS="autoconf file g++ gcc libc-dev pkgconf re2c"
+RUN --mount=type=cache,target=/var/cache/apk,sharing=locked,id=apk \
+  apk add --no-cache $LIB_DEPS $BUILD_DEPS $TOOL_DEPS
 
-FROM phpbase as toolbox
+FROM --platform=${BUILDPLATFORM} alpine AS ast-downloader
```

Comment:
`BUILDPLATFORM` here means it runs using the native architecture (because the git clone doesn't care about CPU)